src/F1_plotting.py

```diff
@@ -171,95 +162,220 @@ def parse_simulation_log(path_log_file=None, log_type="ERROR"):
+
+    # I if the last line is not empty
+    if n_lines > 0:
+        if last_line_length > 0:
+            split_text.append(text[((i + 1) * char_num) :])
+        answer = "\n".join(split_text)
+    else:
+        answer = text
+    return answer
+
+
+class ESGraph:
+    def __init__(
+        self,
+        energy_system=None,
+        filepath="network",
+        img_format=None,
+        legend=True,
+        txt_width=10,
+        **kwargs
+    ):
+        """Draw the energy system with Graphviz.
+
+        Parameters
+        ----------
+        energy_system: `oemof.solph.network.EnergySystem`
+            The oemof energy stystem
+
+        filepath: str
+            path, where the rendered result shall be saved, if an extension is provided, the format
+            will be automatically adapted except if the `img_format` argument is provided
+            Default: "network"
+
+        img_format: str
+            extension of the available image formats of graphviz (e.g "png", "svg", "pdf", ... )
+            Default: "pdf"
+
+        legend: bool
+            specify, whether a legend will be added to the graph or not
+            Default: False
+
+        Returns
+        -------
+        None: render the generated dot graph in the filepath
+        """
+        file_name, file_ext = os.path.splitext(filepath)
+
+        if img_format is None:
+            if file_ext != "":
+                img_format = file_ext.replace(".", "")
+            else:
+                img_format = "pdf"
+
+        self.dot = graphviz.Digraph(filename=file_name, format=img_format, **kwargs)
+        self.txt_width = txt_width
+        self.busses = []
+
+        if legend is True:
+            with self.dot.subgraph(name="cluster_1") as c:
+                # color of the legend box
+                c.attr(color="black")
+                # title of the legend box
+                c.attr(label="Legends")
+                self.add_bus(subgraph=c)
+                self.add_sink(subgraph=c)
+                self.add_source(subgraph=c)
+                self.add_transformer(subgraph=c)
+                self.add_storage(subgraph=c)
+
+        # draw a node for each of the network's component. The shape depends on the component's type
+        for nd in energy_system.nodes:
+            if isinstance(nd, oemof.solph.network.Bus):
+                self.add_bus(nd.label)
+                # keep the bus reference for drawing edges later
+                self.busses.append(nd)
+            if isinstance(nd, oemof.solph.network.Sink):
+                self.add_sink(nd.label)
+
+            if isinstance(nd, oemof.solph.network.Source):
+                self.add_source(nd.label)
+
+            if isinstance(nd, oemof.solph.network.Transformer):
+                self.add_transformer(nd.label)
+
+            if isinstance(nd, oemof.solph.components.GenericStorage):
+                self.add_storage(nd.label)
```

Comment:
Why are you using `if-if`alyways and not `if-elif`? I would change to `if-elif`, because we require an `if-elif-else` clause here: oemof-solph also has other assets apart from the above, and we need an error message telling us to update the if clauses to render the newly added types.

Reply:
good point

Reply:
Done in 547d9ba817a2e3a17382e5e05280627620fb2503